src/bluetooth/BreachManager.js

```diff
@@ -108,32 +108,52 @@ class BreachManager {
     return !(temperature >= thresholdMinTemperature && temperature <= thresholdMaxTemperature);
   };
 
-  couldBeInBreach = (log, configs) =>
-    configs.some(
+  getConfigInBreach = (log, configs) =>
+    configs.find(
       config =>
         log.temperature >= config.minimumTemperature && log.temperature <= config.maximumTemperature
     );
 
   createBreaches = (sensor, logs, configs, mostRecentBreach) => {
-    let potentialBreach = [];
+    const potentialBreach = { configInBreach: null, logs: [] };
     let currentBreach = mostRecentBreach;
 
     const breaches = mostRecentBreach && !mostRecentBreach.endTimestamp ? [mostRecentBreach] : [];
     const temperatureLogs = [];
 
     logs.forEach(log => {
-      const couldBeInBreach = this.couldBeInBreach(log, configs);
+      const configInBreach = this.getConfigInBreach(log, configs);
       const willCloseBreach = this.willCloseBreach(currentBreach, log);
       const willContinueBreach = this.willContinueBreach(currentBreach, log);
 
       if (willCloseBreach) {
         this.closeBreach(currentBreach, log.timestamp);
         currentBreach = null;
-        potentialBreach = [];
+        potentialBreach.logs = [];
+        potentialBreach.configInBreach = null;
       }
 
-      if (couldBeInBreach) potentialBreach.push(log);
-      else potentialBreach = [];
+      if (configInBreach) {
+        // Assign a config to the stack of logs
+        if (!potentialBreach.configInBreach && !logs) {
```

Comment:
Is `logs` always at least an empty array?

Reply:
Oops, yes it is. I think this block isn't actually needed as the calculations are still working with this never getting executed 🤦‍♀️ . Will remove

Reply:
ccddaedf